Packs/CommonScripts/Scripts/GetEndpointData/GetEndpointData.py

```diff
@@ -0,0 +1,719 @@
+                A tuple containing:
+                - A list of command context outputs.
+                - A human-readable string of the results.
+                - A list of command error outputs.
+        """
+
+        command_context_outputs = []
+        human_readable_outputs = []
+        command_error_outputs = []
+        demisto.debug(f'get_commands_outputs for command "{command}" with result {results}')
+
+        for entry in results:
+            demisto.debug(f'entry {json.dumps(entry)}')
+            command_context_outputs.append(entry.get("EntryContext", {}))
+            if is_error(entry):
+                command_error_outputs.extend(
+                    prepare_human_readable(
+                        command, args, get_error(entry), is_error=True
+                    )
+                )
+            else:
+                human_readable_outputs.append(entry.get("HumanReadable") or "")
+
+        human_readable = "\n".join(human_readable_outputs)
+        command_outputs = {
+            'context': command_context_outputs,
+            'hr': human_readable,
+            'errors': command_error_outputs
+        }
+        demisto.debug(f'{command_outputs=}')
+        return command_context_outputs, human_readable, command_error_outputs
+
+
+def to_list(var):
+    if not var:
+        return []
+    return [var] if not isinstance(var, list) else var
+
+
+def safe_list_get(lst: list, idx: int, default: Any):
+    """
+    Safely retrieves an element from a list at the specified index.
+
+    Args:
+    l (list): The input list.
+    idx (int): The index of the element to retrieve.
+    default (Any): The default value to return if the index is out of range.
+
+    Returns:
+    Any: The element at the specified index if it exists, otherwise the default value.
+    """
+    try:
+        return lst[idx]
+    except IndexError:
+        return default
+
+
+def create_endpoint(
+        command_output: dict[str, Any],
+        output_mapping: dict[str, str],
+        source: str
+) -> dict[str, Any]:
+    """
+    Creates an endpoint dictionary from command output, output mapping, and source.
+
+    This function processes the command output and creates a structured endpoint dictionary.
+    It maps the command output keys to endpoint keys based on the provided output mapping,
+    and includes the source information for each value.
+
+    Args:
+        command_output (dict[str, Any]): The output from a command execution.
+        output_mapping (dict[str, str] | Callable): A mapping of command output keys to endpoint keys.
+            If a function is passed, the function does nothing and returns the result of the passed function.
+        source (str): The source of the data.
+
+    Returns:
+        dict[str, Any]: A structured endpoint dictionary with values and their sources.
+    """
+    demisto.debug(f'creating endpoint with {command_output=}, {output_mapping=}, {source=}')
+
+    if not command_output:
```

Comment:
When the raw endpoint will be `None`?

Reply:
Some commands return None in the endpoint is not found. Other return an empty list of entries.